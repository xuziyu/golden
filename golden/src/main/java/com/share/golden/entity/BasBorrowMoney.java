package com.share.golden.entity;

public class BasBorrowMoney {
    private String borrowId;

    private String borrowSerial;

    private String userId;

    private String borrowCompany;

    private String borrowMoney;

    private String borrowRate;

    private String timeLimit;

    private String borrowLimit;

    private String borrowState;

    private String borrowRecommend;

    private String borrowType;

    private String borrowRemark1;

    private String borrowRemark2;

    public String getBorrowId() {
        return borrowId;
    }

    public void setBorrowId(String borrowId) {
        this.borrowId = borrowId == null ? null : borrowId.trim();
    }

    public String getBorrowSerial() {
        return borrowSerial;
    }

    public void setBorrowSerial(String borrowSerial) {
        this.borrowSerial = borrowSerial == null ? null : borrowSerial.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getBorrowCompany() {
        return borrowCompany;
    }

    public void setBorrowCompany(String borrowCompany) {
        this.borrowCompany = borrowCompany == null ? null : borrowCompany.trim();
    }

    public String getBorrowMoney() {
        return borrowMoney;
    }

    public void setBorrowMoney(String borrowMoney) {
        this.borrowMoney = borrowMoney == null ? null : borrowMoney.trim();
    }

    public String getBorrowRate() {
        return borrowRate;
    }

    public void setBorrowRate(String borrowRate) {
        this.borrowRate = borrowRate == null ? null : borrowRate.trim();
    }

    public String getTimeLimit() {
        return timeLimit;
    }

    public void setTimeLimit(String timeLimit) {
        this.timeLimit = timeLimit == null ? null : timeLimit.trim();
    }

    public String getBorrowLimit() {
        return borrowLimit;
    }

    public void setBorrowLimit(String borrowLimit) {
        this.borrowLimit = borrowLimit == null ? null : borrowLimit.trim();
    }

    public String getBorrowState() {
        return borrowState;
    }

    public void setBorrowState(String borrowState) {
        this.borrowState = borrowState == null ? null : borrowState.trim();
    }

    public String getBorrowRecommend() {
        return borrowRecommend;
    }

    public void setBorrowRecommend(String borrowRecommend) {
        this.borrowRecommend = borrowRecommend == null ? null : borrowRecommend.trim();
    }

    public String getBorrowType() {
        return borrowType;
    }

    public void setBorrowType(String borrowType) {
        this.borrowType = borrowType == null ? null : borrowType.trim();
    }

    public String getBorrowRemark1() {
        return borrowRemark1;
    }

    public void setBorrowRemark1(String borrowRemark1) {
        this.borrowRemark1 = borrowRemark1 == null ? null : borrowRemark1.trim();
    }

    public String getBorrowRemark2() {
        return borrowRemark2;
    }

    public void setBorrowRemark2(String borrowRemark2) {
        this.borrowRemark2 = borrowRemark2 == null ? null : borrowRemark2.trim();
    }
}