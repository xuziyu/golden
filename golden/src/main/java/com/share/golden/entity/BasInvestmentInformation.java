package com.share.golden.entity;

import java.math.BigDecimal;
import java.util.Date;

public class BasInvestmentInformation {
    private String investorId;

    private String userId;

    private String tenderId;

    private BigDecimal investorMoney;

    private String investorStatus;

    private String investorStyle;

    private String brrowStatus;

    private String investorRest;

    private String profitModel;

    private BigDecimal profitMoney;

    private Date investorDate;

    private String replayDate;

    private String investorType;

    public String getInvestorId() {
        return investorId;
    }

    public void setInvestorId(String investorId) {
        this.investorId = investorId == null ? null : investorId.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getTenderId() {
        return tenderId;
    }

    public void setTenderId(String tenderId) {
        this.tenderId = tenderId == null ? null : tenderId.trim();
    }

    public BigDecimal getInvestorMoney() {
        return investorMoney;
    }

    public void setInvestorMoney(BigDecimal investorMoney) {
        this.investorMoney = investorMoney;
    }

    public String getInvestorStatus() {
        return investorStatus;
    }

    public void setInvestorStatus(String investorStatus) {
        this.investorStatus = investorStatus == null ? null : investorStatus.trim();
    }

    public String getInvestorStyle() {
        return investorStyle;
    }

    public void setInvestorStyle(String investorStyle) {
        this.investorStyle = investorStyle == null ? null : investorStyle.trim();
    }

    public String getBrrowStatus() {
        return brrowStatus;
    }

    public void setBrrowStatus(String brrowStatus) {
        this.brrowStatus = brrowStatus == null ? null : brrowStatus.trim();
    }

    public String getInvestorRest() {
        return investorRest;
    }

    public void setInvestorRest(String investorRest) {
        this.investorRest = investorRest == null ? null : investorRest.trim();
    }

    public String getProfitModel() {
        return profitModel;
    }

    public void setProfitModel(String profitModel) {
        this.profitModel = profitModel == null ? null : profitModel.trim();
    }

    public BigDecimal getProfitMoney() {
        return profitMoney;
    }

    public void setProfitMoney(BigDecimal profitMoney) {
        this.profitMoney = profitMoney;
    }

    public Date getInvestorDate() {
        return investorDate;
    }

    public void setInvestorDate(Date investorDate) {
        this.investorDate = investorDate;
    }

    public String getReplayDate() {
        return replayDate;
    }

    public void setReplayDate(String replayDate) {
        this.replayDate = replayDate == null ? null : replayDate.trim();
    }

    public String getInvestorType() {
        return investorType;
    }

    public void setInvestorType(String investorType) {
        this.investorType = investorType == null ? null : investorType.trim();
    }
}