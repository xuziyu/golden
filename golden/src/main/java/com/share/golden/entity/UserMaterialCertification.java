package com.share.golden.entity;

import java.util.Date;

public class UserMaterialCertification {
    private String materiaId;

    private String userId;

    private String materiaType;

    private String materia;

    private Date createDate;

    private String valid;

    public String getMateriaId() {
        return materiaId;
    }

    public void setMateriaId(String materiaId) {
        this.materiaId = materiaId == null ? null : materiaId.trim();
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId == null ? null : userId.trim();
    }

    public String getMateriaType() {
        return materiaType;
    }

    public void setMateriaType(String materiaType) {
        this.materiaType = materiaType == null ? null : materiaType.trim();
    }

    public String getMateria() {
        return materia;
    }

    public void setMateria(String materia) {
        this.materia = materia == null ? null : materia.trim();
    }

    public Date getCreateDate() {
        return createDate;
    }

    public void setCreateDate(Date createDate) {
        this.createDate = createDate;
    }

    public String getValid() {
        return valid;
    }

    public void setValid(String valid) {
        this.valid = valid == null ? null : valid.trim();
    }
}