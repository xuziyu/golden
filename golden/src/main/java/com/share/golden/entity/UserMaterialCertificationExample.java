package com.share.golden.entity;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class UserMaterialCertificationExample {
    protected String orderByClause;

    protected boolean distinct;

    protected List<Criteria> oredCriteria;

    public UserMaterialCertificationExample() {
        oredCriteria = new ArrayList<Criteria>();
    }

    public void setOrderByClause(String orderByClause) {
        this.orderByClause = orderByClause;
    }

    public String getOrderByClause() {
        return orderByClause;
    }

    public void setDistinct(boolean distinct) {
        this.distinct = distinct;
    }

    public boolean isDistinct() {
        return distinct;
    }

    public List<Criteria> getOredCriteria() {
        return oredCriteria;
    }

    public void or(Criteria criteria) {
        oredCriteria.add(criteria);
    }

    public Criteria or() {
        Criteria criteria = createCriteriaInternal();
        oredCriteria.add(criteria);
        return criteria;
    }

    public Criteria createCriteria() {
        Criteria criteria = createCriteriaInternal();
        if (oredCriteria.size() == 0) {
            oredCriteria.add(criteria);
        }
        return criteria;
    }

    protected Criteria createCriteriaInternal() {
        Criteria criteria = new Criteria();
        return criteria;
    }

    public void clear() {
        oredCriteria.clear();
        orderByClause = null;
        distinct = false;
    }

    protected abstract static class GeneratedCriteria {
        protected List<Criterion> criteria;

        protected GeneratedCriteria() {
            super();
            criteria = new ArrayList<Criterion>();
        }

        public boolean isValid() {
            return criteria.size() > 0;
        }

        public List<Criterion> getAllCriteria() {
            return criteria;
        }

        public List<Criterion> getCriteria() {
            return criteria;
        }

        protected void addCriterion(String condition) {
            if (condition == null) {
                throw new RuntimeException("Value for condition cannot be null");
            }
            criteria.add(new Criterion(condition));
        }

        protected void addCriterion(String condition, Object value, String property) {
            if (value == null) {
                throw new RuntimeException("Value for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value));
        }

        protected void addCriterion(String condition, Object value1, Object value2, String property) {
            if (value1 == null || value2 == null) {
                throw new RuntimeException("Between values for " + property + " cannot be null");
            }
            criteria.add(new Criterion(condition, value1, value2));
        }

        public Criteria andMateriaIdIsNull() {
            addCriterion("\"MATERIA_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andMateriaIdIsNotNull() {
            addCriterion("\"MATERIA_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andMateriaIdEqualTo(String value) {
            addCriterion("\"MATERIA_ID\" =", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdNotEqualTo(String value) {
            addCriterion("\"MATERIA_ID\" <>", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdGreaterThan(String value) {
            addCriterion("\"MATERIA_ID\" >", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"MATERIA_ID\" >=", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdLessThan(String value) {
            addCriterion("\"MATERIA_ID\" <", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdLessThanOrEqualTo(String value) {
            addCriterion("\"MATERIA_ID\" <=", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdLike(String value) {
            addCriterion("\"MATERIA_ID\" like", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdNotLike(String value) {
            addCriterion("\"MATERIA_ID\" not like", value, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdIn(List<String> values) {
            addCriterion("\"MATERIA_ID\" in", values, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdNotIn(List<String> values) {
            addCriterion("\"MATERIA_ID\" not in", values, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdBetween(String value1, String value2) {
            addCriterion("\"MATERIA_ID\" between", value1, value2, "materiaId");
            return (Criteria) this;
        }

        public Criteria andMateriaIdNotBetween(String value1, String value2) {
            addCriterion("\"MATERIA_ID\" not between", value1, value2, "materiaId");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNull() {
            addCriterion("\"USER_ID\" is null");
            return (Criteria) this;
        }

        public Criteria andUserIdIsNotNull() {
            addCriterion("\"USER_ID\" is not null");
            return (Criteria) this;
        }

        public Criteria andUserIdEqualTo(String value) {
            addCriterion("\"USER_ID\" =", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotEqualTo(String value) {
            addCriterion("\"USER_ID\" <>", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThan(String value) {
            addCriterion("\"USER_ID\" >", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdGreaterThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" >=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThan(String value) {
            addCriterion("\"USER_ID\" <", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLessThanOrEqualTo(String value) {
            addCriterion("\"USER_ID\" <=", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdLike(String value) {
            addCriterion("\"USER_ID\" like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotLike(String value) {
            addCriterion("\"USER_ID\" not like", value, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdIn(List<String> values) {
            addCriterion("\"USER_ID\" in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotIn(List<String> values) {
            addCriterion("\"USER_ID\" not in", values, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andUserIdNotBetween(String value1, String value2) {
            addCriterion("\"USER_ID\" not between", value1, value2, "userId");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeIsNull() {
            addCriterion("\"MATERIA_TYPE\" is null");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeIsNotNull() {
            addCriterion("\"MATERIA_TYPE\" is not null");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeEqualTo(String value) {
            addCriterion("\"MATERIA_TYPE\" =", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeNotEqualTo(String value) {
            addCriterion("\"MATERIA_TYPE\" <>", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeGreaterThan(String value) {
            addCriterion("\"MATERIA_TYPE\" >", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeGreaterThanOrEqualTo(String value) {
            addCriterion("\"MATERIA_TYPE\" >=", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeLessThan(String value) {
            addCriterion("\"MATERIA_TYPE\" <", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeLessThanOrEqualTo(String value) {
            addCriterion("\"MATERIA_TYPE\" <=", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeLike(String value) {
            addCriterion("\"MATERIA_TYPE\" like", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeNotLike(String value) {
            addCriterion("\"MATERIA_TYPE\" not like", value, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeIn(List<String> values) {
            addCriterion("\"MATERIA_TYPE\" in", values, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeNotIn(List<String> values) {
            addCriterion("\"MATERIA_TYPE\" not in", values, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeBetween(String value1, String value2) {
            addCriterion("\"MATERIA_TYPE\" between", value1, value2, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaTypeNotBetween(String value1, String value2) {
            addCriterion("\"MATERIA_TYPE\" not between", value1, value2, "materiaType");
            return (Criteria) this;
        }

        public Criteria andMateriaIsNull() {
            addCriterion("\"MATERIA\" is null");
            return (Criteria) this;
        }

        public Criteria andMateriaIsNotNull() {
            addCriterion("\"MATERIA\" is not null");
            return (Criteria) this;
        }

        public Criteria andMateriaEqualTo(String value) {
            addCriterion("\"MATERIA\" =", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaNotEqualTo(String value) {
            addCriterion("\"MATERIA\" <>", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaGreaterThan(String value) {
            addCriterion("\"MATERIA\" >", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaGreaterThanOrEqualTo(String value) {
            addCriterion("\"MATERIA\" >=", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaLessThan(String value) {
            addCriterion("\"MATERIA\" <", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaLessThanOrEqualTo(String value) {
            addCriterion("\"MATERIA\" <=", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaLike(String value) {
            addCriterion("\"MATERIA\" like", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaNotLike(String value) {
            addCriterion("\"MATERIA\" not like", value, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaIn(List<String> values) {
            addCriterion("\"MATERIA\" in", values, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaNotIn(List<String> values) {
            addCriterion("\"MATERIA\" not in", values, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaBetween(String value1, String value2) {
            addCriterion("\"MATERIA\" between", value1, value2, "materia");
            return (Criteria) this;
        }

        public Criteria andMateriaNotBetween(String value1, String value2) {
            addCriterion("\"MATERIA\" not between", value1, value2, "materia");
            return (Criteria) this;
        }

        public Criteria andCreateDateIsNull() {
            addCriterion("\"CREATE_DATE\" is null");
            return (Criteria) this;
        }

        public Criteria andCreateDateIsNotNull() {
            addCriterion("\"CREATE_DATE\" is not null");
            return (Criteria) this;
        }

        public Criteria andCreateDateEqualTo(Date value) {
            addCriterion("\"CREATE_DATE\" =", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateNotEqualTo(Date value) {
            addCriterion("\"CREATE_DATE\" <>", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateGreaterThan(Date value) {
            addCriterion("\"CREATE_DATE\" >", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateGreaterThanOrEqualTo(Date value) {
            addCriterion("\"CREATE_DATE\" >=", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateLessThan(Date value) {
            addCriterion("\"CREATE_DATE\" <", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateLessThanOrEqualTo(Date value) {
            addCriterion("\"CREATE_DATE\" <=", value, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateIn(List<Date> values) {
            addCriterion("\"CREATE_DATE\" in", values, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateNotIn(List<Date> values) {
            addCriterion("\"CREATE_DATE\" not in", values, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateBetween(Date value1, Date value2) {
            addCriterion("\"CREATE_DATE\" between", value1, value2, "createDate");
            return (Criteria) this;
        }

        public Criteria andCreateDateNotBetween(Date value1, Date value2) {
            addCriterion("\"CREATE_DATE\" not between", value1, value2, "createDate");
            return (Criteria) this;
        }

        public Criteria andValidIsNull() {
            addCriterion("\"VALID\" is null");
            return (Criteria) this;
        }

        public Criteria andValidIsNotNull() {
            addCriterion("\"VALID\" is not null");
            return (Criteria) this;
        }

        public Criteria andValidEqualTo(String value) {
            addCriterion("\"VALID\" =", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidNotEqualTo(String value) {
            addCriterion("\"VALID\" <>", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidGreaterThan(String value) {
            addCriterion("\"VALID\" >", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidGreaterThanOrEqualTo(String value) {
            addCriterion("\"VALID\" >=", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidLessThan(String value) {
            addCriterion("\"VALID\" <", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidLessThanOrEqualTo(String value) {
            addCriterion("\"VALID\" <=", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidLike(String value) {
            addCriterion("\"VALID\" like", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidNotLike(String value) {
            addCriterion("\"VALID\" not like", value, "valid");
            return (Criteria) this;
        }

        public Criteria andValidIn(List<String> values) {
            addCriterion("\"VALID\" in", values, "valid");
            return (Criteria) this;
        }

        public Criteria andValidNotIn(List<String> values) {
            addCriterion("\"VALID\" not in", values, "valid");
            return (Criteria) this;
        }

        public Criteria andValidBetween(String value1, String value2) {
            addCriterion("\"VALID\" between", value1, value2, "valid");
            return (Criteria) this;
        }

        public Criteria andValidNotBetween(String value1, String value2) {
            addCriterion("\"VALID\" not between", value1, value2, "valid");
            return (Criteria) this;
        }
    }

    public static class Criteria extends GeneratedCriteria {

        protected Criteria() {
            super();
        }
    }

    public static class Criterion {
        private String condition;

        private Object value;

        private Object secondValue;

        private boolean noValue;

        private boolean singleValue;

        private boolean betweenValue;

        private boolean listValue;

        private String typeHandler;

        public String getCondition() {
            return condition;
        }

        public Object getValue() {
            return value;
        }

        public Object getSecondValue() {
            return secondValue;
        }

        public boolean isNoValue() {
            return noValue;
        }

        public boolean isSingleValue() {
            return singleValue;
        }

        public boolean isBetweenValue() {
            return betweenValue;
        }

        public boolean isListValue() {
            return listValue;
        }

        public String getTypeHandler() {
            return typeHandler;
        }

        protected Criterion(String condition) {
            super();
            this.condition = condition;
            this.typeHandler = null;
            this.noValue = true;
        }

        protected Criterion(String condition, Object value, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.typeHandler = typeHandler;
            if (value instanceof List<?>) {
                this.listValue = true;
            } else {
                this.singleValue = true;
            }
        }

        protected Criterion(String condition, Object value) {
            this(condition, value, null);
        }

        protected Criterion(String condition, Object value, Object secondValue, String typeHandler) {
            super();
            this.condition = condition;
            this.value = value;
            this.secondValue = secondValue;
            this.typeHandler = typeHandler;
            this.betweenValue = true;
        }

        protected Criterion(String condition, Object value, Object secondValue) {
            this(condition, value, secondValue, null);
        }
    }
}