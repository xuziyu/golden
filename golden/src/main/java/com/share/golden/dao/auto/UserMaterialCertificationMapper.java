package com.share.golden.dao.auto;

import com.share.golden.entity.UserMaterialCertification;
import com.share.golden.entity.UserMaterialCertificationExample;
import java.util.List;
import org.apache.ibatis.annotations.Param;

public interface UserMaterialCertificationMapper {
    long countByExample(UserMaterialCertificationExample example);

    int deleteByExample(UserMaterialCertificationExample example);

    int deleteByPrimaryKey(String materiaId);

    int insert(UserMaterialCertification record);

    int insertSelective(UserMaterialCertification record);

    List<UserMaterialCertification> selectByExample(UserMaterialCertificationExample example);

    UserMaterialCertification selectByPrimaryKey(String materiaId);

    int updateByExampleSelective(@Param("record") UserMaterialCertification record, @Param("example") UserMaterialCertificationExample example);

    int updateByExample(@Param("record") UserMaterialCertification record, @Param("example") UserMaterialCertificationExample example);

    int updateByPrimaryKeySelective(UserMaterialCertification record);

    int updateByPrimaryKey(UserMaterialCertification record);
}