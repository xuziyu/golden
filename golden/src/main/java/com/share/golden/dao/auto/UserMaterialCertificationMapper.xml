<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE mapper PUBLIC "-//mybatis.org//DTD Mapper 3.0//EN" "http://mybatis.org/dtd/mybatis-3-mapper.dtd">
<mapper namespace="com.share.golden.dao.auto.UserMaterialCertificationMapper">
  <resultMap id="BaseResultMap" type="com.share.golden.entity.UserMaterialCertification">
    <id column="MATERIA_ID" jdbcType="VARCHAR" property="materiaId" />
    <result column="USER_ID" jdbcType="VARCHAR" property="userId" />
    <result column="MATERIA_TYPE" jdbcType="VARCHAR" property="materiaType" />
    <result column="MATERIA" jdbcType="VARCHAR" property="materia" />
    <result column="CREATE_DATE" jdbcType="TIMESTAMP" property="createDate" />
    <result column="VALID" jdbcType="VARCHAR" property="valid" />
  </resultMap>
  <sql id="Example_Where_Clause">
    <where>
      <foreach collection="oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Update_By_Example_Where_Clause">
    <where>
      <foreach collection="example.oredCriteria" item="criteria" separator="or">
        <if test="criteria.valid">
          <trim prefix="(" prefixOverrides="and" suffix=")">
            <foreach collection="criteria.criteria" item="criterion">
              <choose>
                <when test="criterion.noValue">
                  and ${criterion.condition}
                </when>
                <when test="criterion.singleValue">
                  and ${criterion.condition} #{criterion.value}
                </when>
                <when test="criterion.betweenValue">
                  and ${criterion.condition} #{criterion.value} and #{criterion.secondValue}
                </when>
                <when test="criterion.listValue">
                  and ${criterion.condition}
                  <foreach close=")" collection="criterion.value" item="listItem" open="(" separator=",">
                    #{listItem}
                  </foreach>
                </when>
              </choose>
            </foreach>
          </trim>
        </if>
      </foreach>
    </where>
  </sql>
  <sql id="Base_Column_List">
    "MATERIA_ID", "USER_ID", "MATERIA_TYPE", "MATERIA", "CREATE_DATE", "VALID"
  </sql>
  <select id="selectByExample" parameterType="com.share.golden.entity.UserMaterialCertificationExample" resultMap="BaseResultMap">
    select
    <if test="distinct">
      distinct
    </if>
    <include refid="Base_Column_List" />
    from USER_MATERIAL_CERTIFICATION
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
    <if test="orderByClause != null">
      order by ${orderByClause}
    </if>
  </select>
  <select id="selectByPrimaryKey" parameterType="java.lang.String" resultMap="BaseResultMap">
    select 
    <include refid="Base_Column_List" />
    from USER_MATERIAL_CERTIFICATION
    where "MATERIA_ID" = #{materiaId,jdbcType=VARCHAR}
  </select>
  <delete id="deleteByPrimaryKey" parameterType="java.lang.String">
    delete from USER_MATERIAL_CERTIFICATION
    where "MATERIA_ID" = #{materiaId,jdbcType=VARCHAR}
  </delete>
  <delete id="deleteByExample" parameterType="com.share.golden.entity.UserMaterialCertificationExample">
    delete from USER_MATERIAL_CERTIFICATION
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </delete>
  <insert id="insert" parameterType="com.share.golden.entity.UserMaterialCertification">
    insert into USER_MATERIAL_CERTIFICATION ("MATERIA_ID", "USER_ID", "MATERIA_TYPE", 
      "MATERIA", "CREATE_DATE", "VALID"
      )
    values (#{materiaId,jdbcType=VARCHAR}, #{userId,jdbcType=VARCHAR}, #{materiaType,jdbcType=VARCHAR}, 
      #{materia,jdbcType=VARCHAR}, #{createDate,jdbcType=TIMESTAMP}, #{valid,jdbcType=VARCHAR}
      )
  </insert>
  <insert id="insertSelective" parameterType="com.share.golden.entity.UserMaterialCertification">
    insert into USER_MATERIAL_CERTIFICATION
    <trim prefix="(" suffix=")" suffixOverrides=",">
      <if test="materiaId != null">
        "MATERIA_ID",
      </if>
      <if test="userId != null">
        "USER_ID",
      </if>
      <if test="materiaType != null">
        "MATERIA_TYPE",
      </if>
      <if test="materia != null">
        "MATERIA",
      </if>
      <if test="createDate != null">
        "CREATE_DATE",
      </if>
      <if test="valid != null">
        "VALID",
      </if>
    </trim>
    <trim prefix="values (" suffix=")" suffixOverrides=",">
      <if test="materiaId != null">
        #{materiaId,jdbcType=VARCHAR},
      </if>
      <if test="userId != null">
        #{userId,jdbcType=VARCHAR},
      </if>
      <if test="materiaType != null">
        #{materiaType,jdbcType=VARCHAR},
      </if>
      <if test="materia != null">
        #{materia,jdbcType=VARCHAR},
      </if>
      <if test="createDate != null">
        #{createDate,jdbcType=TIMESTAMP},
      </if>
      <if test="valid != null">
        #{valid,jdbcType=VARCHAR},
      </if>
    </trim>
  </insert>
  <select id="countByExample" parameterType="com.share.golden.entity.UserMaterialCertificationExample" resultType="java.lang.Long">
    select count(*) from USER_MATERIAL_CERTIFICATION
    <if test="_parameter != null">
      <include refid="Example_Where_Clause" />
    </if>
  </select>
  <update id="updateByExampleSelective" parameterType="map">
    update USER_MATERIAL_CERTIFICATION
    <set>
      <if test="record.materiaId != null">
        "MATERIA_ID" = #{record.materiaId,jdbcType=VARCHAR},
      </if>
      <if test="record.userId != null">
        "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      </if>
      <if test="record.materiaType != null">
        "MATERIA_TYPE" = #{record.materiaType,jdbcType=VARCHAR},
      </if>
      <if test="record.materia != null">
        "MATERIA" = #{record.materia,jdbcType=VARCHAR},
      </if>
      <if test="record.createDate != null">
        "CREATE_DATE" = #{record.createDate,jdbcType=TIMESTAMP},
      </if>
      <if test="record.valid != null">
        "VALID" = #{record.valid,jdbcType=VARCHAR},
      </if>
    </set>
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByExample" parameterType="map">
    update USER_MATERIAL_CERTIFICATION
    set "MATERIA_ID" = #{record.materiaId,jdbcType=VARCHAR},
      "USER_ID" = #{record.userId,jdbcType=VARCHAR},
      "MATERIA_TYPE" = #{record.materiaType,jdbcType=VARCHAR},
      "MATERIA" = #{record.materia,jdbcType=VARCHAR},
      "CREATE_DATE" = #{record.createDate,jdbcType=TIMESTAMP},
      "VALID" = #{record.valid,jdbcType=VARCHAR}
    <if test="_parameter != null">
      <include refid="Update_By_Example_Where_Clause" />
    </if>
  </update>
  <update id="updateByPrimaryKeySelective" parameterType="com.share.golden.entity.UserMaterialCertification">
    update USER_MATERIAL_CERTIFICATION
    <set>
      <if test="userId != null">
        "USER_ID" = #{userId,jdbcType=VARCHAR},
      </if>
      <if test="materiaType != null">
        "MATERIA_TYPE" = #{materiaType,jdbcType=VARCHAR},
      </if>
      <if test="materia != null">
        "MATERIA" = #{materia,jdbcType=VARCHAR},
      </if>
      <if test="createDate != null">
        "CREATE_DATE" = #{createDate,jdbcType=TIMESTAMP},
      </if>
      <if test="valid != null">
        "VALID" = #{valid,jdbcType=VARCHAR},
      </if>
    </set>
    where "MATERIA_ID" = #{materiaId,jdbcType=VARCHAR}
  </update>
  <update id="updateByPrimaryKey" parameterType="com.share.golden.entity.UserMaterialCertification">
    update USER_MATERIAL_CERTIFICATION
    set "USER_ID" = #{userId,jdbcType=VARCHAR},
      "MATERIA_TYPE" = #{materiaType,jdbcType=VARCHAR},
      "MATERIA" = #{materia,jdbcType=VARCHAR},
      "CREATE_DATE" = #{createDate,jdbcType=TIMESTAMP},
      "VALID" = #{valid,jdbcType=VARCHAR}
    where "MATERIA_ID" = #{materiaId,jdbcType=VARCHAR}
  </update>
</mapper>